File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/runningkey/RunningKeyKey.kt
package cryptobench.ciphers.runningkey

data class RunningKeyKey(val stream: String) {

    fun keyCharAt(position: Int): Char = stream[position % stream.length]

    companion object {
        private const val PASSAGE =
            "ITWASABRIGHTCOLDDAYINAPRILANDTHECLOCKSWERESTRIKINGTHIRTEEN"

        fun default(): RunningKeyKey = RunningKeyKey(PASSAGE)
    }
}

File: fixtures/scale/dart/large/lib/ciphers/runningkey/runningkey_key.dart
const _passage = 'ITWASABRIGHTCOLDDAYINAPRILANDTHECLOCKSWERESTRIKINGTHIRTEEN';

class RunningKeyKey {
  final String stream;

  RunningKeyKey(this.stream);

  String keyCharAt(int position) => stream[position % stream.length];

  factory RunningKeyKey.defaultKey() => RunningKeyKey(_passage);
}
